#AMF Florida and Califronia Samples Celia Xi and Zella D. 

```{r}

#Phyloseq Analysis 


```{r}
#BiocManager::install("UpSetR")
#install.packages("phyloseq")
citation("UpSetR")
sessionInfo()
library("DESeq2")
library(circlize)
#library(ComplexHeatmap)
#library(Polychrome)
library(RColorBrewer)
library("phyloseq")
#library("plyr")
library("scales")
library(ggpubr)
library("VennDiagram")
library(ggplot2)
require(devtools)
#install_version("vegan", version ="2.4-5", repos = "http://cran.us.r-project.org")
library(vegan)
library(reshape)
library(igraph)
library('dada2')
library('phangorn')
library("DECIPHER")
library("dplyr")

library('microbiome')
library('seqinr')
library(tidyverse)
library("RVAideMemoire")
```
#Load DataSets

```{r}
#Load Sequence Tables 

#California 
#cali<-readRDS("/home/ella/Documents/Ella/Celia/AMF_Cali8.17.2020/results/seqtab.nochim.rds")
#flor<-readRDS("/home/ella/Documents/Ella/Celia/AMF_florida/resultsFOnly/seqtab.nochim.rds")
#dim(cali)
#dim(flor)


#intersect(rownames(cali), rownames(flor))
#rownames(cali) 
#rownames(flor)
```
#Fix Rownames for Florida samples
```{r}

#florrn<-sapply(strsplit(rownames(flor), ".fastq"), '[',1)
#florrn<-sapply(strsplit(florrn, "pair1_"), '[', 2)
#rownames(flor)<-florrn

```
#Merge Sequence tables 
```{r}
#amf<-mergeSequenceTables(table1=cali, table2=flor)
#dim(amf)
#class(amf)
#rownames(amf)

#saveRDS(amf, "amf_mergetables8.18.2020.rds")
```

#Assign Taxonomy 
```{r}
#table(nchar(getSequences(amf)))
#seqs<-getSequences(amf)
#names(amf)<-seqs
#writeFasta(seqs, "amfseqs.fasta")
#silva18<-"/home/ella/Documents/Ella/silva_132.18s.99_rep_set.dada2.fa"

#tax<-assignTaxonomy(amf, refFasta=silva18, multithread=TRUE, minBoot=40,
#	taxLevels=c("Superkingdom", "subphylum", "otherthing", "Kingdom", "Class", "Order", "Family", "Genus")
		)
#taxa.print<-data.frame(tax)
#colnames(taxa.print)
#sum(is.na(taxa.print[,7]))
#table(taxa.print[,2])
#table(taxa.print[,4])
#head(taxa.print)
#unique(taxa.print[,6])
#table(taxa.print[, 5])

```
#Add Virtual Taxa from Maarjam
```{r}
#Vts<-read.csv("blastn_VTassignments_565ASV_8.18.2020.csv")
#dim(Vts)
#dim(tax)

#rownames(Vts)<-Vts[,3]
#colnames(Vts)
#Vts[, "VT"]<-as.character(Vts$VT)
#tax2<-merge(taxa.print, Vts, by=0, all=TRUE)
#class(tax2)
#table(tax2[,"Kingdom"])
#table(tax2[,"Phylum"]
```
#Convert to Phyloseq Object
```{r}
#Make sure sample names and tax sequences are aligning between all pieces
#map<-read.csv("AMF_masta_map.csv", stringsAsFactors=T)
#rownames(map)<-map$Sample.ID.

#setdiff(rownames(amf),rownames(map))


#setdiff(colnames(amf), rownames(tax2))
#Want these to be 0. Meaning there is no difference between the respected rownames and colnames

#rownames(tax2)<-tax2[,1]
#colnames(tax2)

#colnames(tax2)
#head(rownames(tax2))
#tax2<-as.matrix(tax2)
#head(tax2)
#sort(as.character(tax2[,3]))

#ps<-phyloseq(otu_table(amf, taxa_are_rows=FALSE),tax_table(tax2), sample_data(map))
#saveRDS(ps,"phyloseq_raw_AMF_FULLVT.8.21.2020.rds")
ps<-readRDS( "phyloseq_raw_AMF_FULLVT.8.21.2020.rds")
ps

taxtabPs<-data.frame(tax_table(ps))
dim(taxtabPs)

codnames<-read.csv("CodeNamesFromTree.11.24.2020.csv")
colnames(codnames)
codnames<-codnames[codnames$CodeName!="NS",c("VT", "CodeName")]
codnames$CodeName<-factor(str_replace(codnames$CodeName, "DK_301", "Dk_301"))

dim(codnames)
rownames(codnames)<-codnames[,1]
levels(codnames$CodeName)
head(rownames(taxtabPs))
intersect(colnames(taxtabPs), colnames(codnames))
taxtabPs<-full_join(taxtabPs, codnames)

rownames(taxtabPs)<-taxtabPs[,1]
setdiff(rownames(taxtabPs), taxa_names(ps))
tax_table(ps)<-as.matrix(taxtabPs)
ps


#Numeric Taxa(>3) and Sample (>500) Filtering
```{r}
#Prune Out samples with less then 500 Taxa
sort(sample_sums(ps))[1:15]
summary(sample_sums(ps))
amf_ps<-subset_samples(ps, sample_sums(ps)>=1000)
summary(sample_sums(amf_ps))
#What's Left
amf_ps

map<-data.frame(sample_data(amf_ps))
map$DiseaseRating<-round(as.numeric(as.character(map$DiseaseRating))-.1,0)
sample_data(amf_ps)<-map
```

#Family Number 

```{r}
citation()
#Quick look at samples
citation("vegan")
amf<-amf_ps
summary(taxa_sums(amf))

#For Tree Taxa >1%
amf<-subset_samples(amf, UseScale!=3 & UseScale!=4)
amf<-subset_taxa(amf, Kingdom=="Fungi")
summary(sample_sums(amf))
amf<-subset_taxa(amf, taxa_sums(amf)>=2)

write.csv(data.frame(sample_data(amf)), "Samples_used_inAMF_Ms_12.1.2020.csv")
amf
amf3<-amf

tax_table(amf3)<-tax_table(amf3)[,c("Kingdom","VT", "CodeName")]

amf3<-tax_glom(amf3, "VT")
amf3
#treVt<-subset_taxa(amf3, taxa_sums(amf3)>20)
#treVt
#tx_tb<-data.frame(tax_table(treVt))
#seqs_tre<-getSequences(rownames(tx_tb))
#names(seqs_tre)<-paste0(tx_tb$Order, "_", tx_tb$VT)
#library('tigger')
#writeFasta(seqs_tre, "VTseqs_min20_forTreeID_11.20.2020.fasta")
sort(names(seqs_tre))

```

#Tax Glom at family
```{r}

amf
map<-data.frame(sample_data(amf))


g<-"graphs_11.20.2020FINAL/"

themesizes<-theme(plot.title=element_text(hjust=0.4, size=16, face="bold"),
                        axis.text.x=element_text(angle=90, hjust=1,vjust=.4, size=12),
                        axis.text.y=element_text(size=12))
cols<-c(brewer.pal(n=5, "PuRd"), brewer.pal(n=3, "BrBG"), brewer.pal(n=10, "RdYlGn") , 
		brewer.pal(n=3, "Dark2"), brewer.pal(n=4, "Accent"))
tax_col<-createPalette(63, "#5ab4ac")
taxnames<-names(rev(sort(table(tax_table(amf)[,"VT"]))))
names(tax_col)<-taxnames

show_col(cols)

summary(sample_data(amf))

#Alpha Diversity main concerns (Between States, Between DR, Between ManagementStrat+Glom level)
states<-subset_samples(amf, UseScale!=3 & UseScale!=4)

DR<-subset_samples(amf, State=="Florida")
DR<-subset_samples(DR, UseScale==2)
mapDR<-data.frame(sample_data(DR))
mapDR$DiseaseRating<-as.factor(mapDR$DiseaseRating)
levels(mapDR$DiseaseRating)
sample_data(DR)<-mapDR
mang<-subset_samples(amf, State=="California")

summary(sample_data(states))
summary(sample_data(DR))
summary(sample_data(mang))
subset_taxa(states, taxa_sums(states)>0)
subset_taxa(amf, taxa_sums(amf)>0)
subset_taxa(DR, taxa_sums(DR)>0)
subset_taxa(mang, taxa_sums(mang)>0)

citation('UpsetR')
#Rarified Data
rar_states<-rarefy_even_depth(states, rngseed=42)
rar_states
summary(sample_sums(rar_states))
rar_mang<-rarefy_even_depth(mang, rngseed=42)
rar_DR<-rarefy_even_depth(DR, rngseed=42)

#Alpha Diversity Observed  + GLM Poisson Tukey Stats
state_a<-plot_richness(states, x="State", measures=c("Observed"))+
	geom_boxplot(aes(fill=State))+labs(x="")+
	theme_classic()+ggtitle('')+themesizes+scale_fill_manual(values=cols[c(6,8)])+
	theme(legend.position='none')

state_a

state_rarA<-plot_richness(rar_states, x="State", measures=c("Observed"))+
	geom_boxplot(aes(fill=State))+labs(x="", y="Observed")+
	theme_classic()+ggtitle('')+themesizes+scale_fill_manual(values=cols[c(6,8)])+
	theme(legend.position='none')

ggarrange(state_a, state_rarA)
set.seed(42)
library(multcomp)

	dat<-states
	obs<-estimate_richness(dat, measures="Observed")
	map<-data.frame(sample_data(dat))
	obs<<-cbind(map, obs)
	print(dim(obs))

	#GLM
	resA_stat<-glm(Observed~State, family=poisson, data=obs)
	#TUKEY	
	resA_stat_sum<-summary(glht(resA_stat, linfct=mcp(State="Tukey")))
	#Results
	print(resA_stat_sum)
packageVersion("vegan")	
#Rarify Stat
packageVersion("vegan")	
#Rarify Stat
	obs<-estimate_richness(rar_states, measures="Observed")
	obs<-cbind(map, obs)	
	resA_stat_rar<-glm(Observed~State, family=poisson, data=obs)
	summary(glht(resA_stat_rar, linfct=mcp(State="Tukey")))
#ggsave(paste0(g, "OBs_shan_state.png"), device='png')

#Alpha Diversity Graphs for Flor(DR) 
  drmap<-data.frame(sample_data(DR))

DR_a<-plot_richness(DR, x="DiseaseRating", measures=c("Observed"))+
	geom_boxplot(aes(fill=DiseaseRating))+
	theme_classic()+ggtitle('')+themesizes+
#geom_point(aes(colour=UseScale,group=UseScale),size=2, alpha=0.5) +
#geom_line(aes(group=Tree), colour="red", linetype="11") +
	scale_fill_manual(values=rev(cols[c(seq(9,18,2))]))+
		labs(x="HLB Disease Rating", y="Observed Taxa")
DR_a
rar_states
DR_rarA<-plot_richness(rar_DR, x="DiseaseRating", measures=c("Observed"))+
	geom_boxplot(aes(fill=DiseaseRating))+labs(x="HLB Disease Rating", y="Observed")+
	theme_classic()+ggtitle('')+themesizes+scale_fill_manual(values=rev(cols[c(seq(9,18,2))]))+
	theme(legend.position='none')
DR_rarA<-DR_rarA+annotate(geom="text",label=c("a", "b", "ab", "c", "b"), x=1:5, y=38, fontface="bold", size=3)
DR_rarA
#Rarify Stat
	obs<-estimate_richness(rar_DR, measures="Observed")
	obs<-cbind(drmap, obs)	
	resA_stat_rar<-glm(Observed~DiseaseRating, family=poisson, data=obs)
	summary(glht(resA_stat_rar, linfct=mcp(DiseaseRating="Tukey")))
ggarrange(DR_a, DR_rarA)
#	ggsave(paste0(g, "Obs_shan_Flor_DR_.png"), device='png')

set.seed(242)
	dat<-DR
	obs<-estimate_richness(dat, measures="Observed")
	map<-data.frame(sample_data(dat))
	obs<-cbind(map, obs)
	#GLM
	resA_stat<-glm(Observed~DiseaseRating, family=poisson, data=obs)
	#TUKEY	
	resA_stat_sum<-summary(glht(resA_stat, linfct=mcp(DiseaseRating="Tukey")))
	#Results
	print(resA_stat_sum)

#Alpha Diversity Graphs for Cali (mang/glom)
mg_a<-plot_richness(mang, x="managment", measures=c("Observed"))+
	geom_boxplot(aes(fill=managment))+
	theme_classic()+ggtitle('')+themesizes+
	scale_fill_manual(values=cols[c(2,5)])+
	theme(legend.position='none')

mg_a

#Rarified
mg_rarA<-plot_richness(rar_mang, x="managment", measures="Observed")+
	geom_boxplot(aes(fill=managment))+
	theme_classic()+ggtitle('')+themesizes+labs(x='', y="Observed")+
	scale_fill_manual(values=cols[c(2,5)])+
	theme(legend.position='none')+annotate(geom="text", label=c("a", "b"), x=1:2, y= 40, 
			fontface="bold", size=3)

ggarrange(mg_a, mg_rarA)
#ggsave(paste0(g, "Obs_shan_Cali_MG.png"), device='png')


 	dat<-mang
	obs<-estimate_richness(dat, measures="Observed")
	map<-data.frame(sample_data(dat))
	obs<-cbind(map, obs)

resA_stat<-glm(Observed~managment+Location, family=poisson, data=obs)
        #TUKEY  
        resA_stat_sum<-summary(glht(resA_stat, linfct=mcp(managment="Tukey")))
        #Results
        print(resA_stat_sum)

	dat<-rar_mang
	obs<-estimate_richness(dat, measures="Observed")
	map<-data.frame(sample_data(dat))
	obs<-cbind(map, obs)

resA_stat<-glm(Observed~managment+Location, family=poisson, data=obs)
        #TUKEY  
        summary(glht(resA_stat, linfct=mcp(managment="Tukey")))


ggarrange(DR_a, labels=c("A"), font.label=list(size=16), 
	ggarrange(state_a, mg_a, nrow=2, labels=c("B", "C")))
#ggsave(paste0(g, "Obs_alphaAll3.png"), device='png')

ggarrange(ggarrange(state_rarA+labs(y="Observed Taxa")+theme(strip.background=element_blank(), strip.text.x=element_blank()), 
		    mg_rarA+labs(y="Observed Taxa")+theme(strip.background=element_blank(), strip.text.x=element_blank()), 
		    nrow=2, labels=c("a", "b"), font.label=list(size=18)), ncol=2,
	  DR_rarA+labs(x="Disease Rating", y="Observed Taxa")+theme(strip.background=element_blank(), strip.text.x=element_blank()), labels=c('','c'), font.label=list(size=18))
ggsave(paste0(g, "Figure2_alpha.png"), device='png', dpi=600, width=80, units='mm', height=200)

ggsave(paste0(g, "Figure2_alpha.tiff"), device='tiff', compression="lzw", dpi=600)


g

```
#Figure 3: Taxa Bar Charts Genus & VT Level
```{r}

#Filtering out unknown phyla and Taxa that occur less then 3 times
library(tidyverse)
amf3
#States

statesvt<-tax_glom(amf3,"VT")
statesvt

statesrav<-transform_sample_counts(statesvt, function(x) x/sum(x))
statesvt

#DiseaseRating
#diseaseg<-tax_glom(DR, "Order")
#diseasev<-subset_samples(amf3, State=="Florida")
#diseasev
##diseaserag<-transform_sample_counts(diseaseg, function(x) x/sum(x))
#diseaserav<-transform_sample_counts(diseasev, function(x) x/sum(x))


#DiseaseRating
#mangmntg<-tax_glom(mang,"Order")
#mangmntv<-subset_samples(amf3, State=="California")

#mangmntrag<-transform_sample_counts(mangmntg, function(x) x/sum(x))
#mangmntrav<-transform_sample_counts(mangmntv, function(x) x/sum(x))



summary(sample_sums(statesrav))

# summary(sample_sums(diseaserav))

#summary(sample_sums(mangmntrav))



#Transform to relabund

colnames(statesrav)
meltmutate<- function(pss, phylo, group){
	psmelted<-psmelt(pss)

	psmelted<-psmelted %>% mutate_if(is.factor,
                      fct_explicit_na,
                      na_level = "(Unknown)")

	colnames(psmelted)
	psmelts<-psmelted%>%group_by({{phylo}}, {{group}},)%>%
                summarize(s=sum(Abundance))%>%arrange(desc(s))
        tots<-psmelts%>%group_by({{group}})%>%summarize(tots=round(sum(s),2))
	psmelts<-merge(psmelts, tots)
	psmelts$RA<-psmelts$s/psmelts$tots
	psmelts<<-psmelts
	print(sum(psmelts$s))
	print(sum(psmelts$RA))
}

#Genus

library(tidyverse)
###################################################
###################VT##############################
psmap<-data.frame(sample_data(statesrav))
psmap$labs<-make.unique(paste0(substr(as.character(psmap$State), 0,2), "_", 
	   ifelse(psmap$DiseaseRating==0, substr(as.character(psmap$managment),0,3), psmap$DiseaseRating)))

psmap$labs
psmap$sample<-"Sample"
psmap$sample
sample_data(statesrav)<-psmap



meltmutate(statesrav, VT, sample)
stmeltsg<-psmelts%>%arrange(desc(RA))

#For Prevt By Sample
meltmutate(statesrav, VT, labs) 
stmetlsP<-psmelts%>%arrange(desc(RA))

stmetlsP$sm_lab<-substr(stmetlsP$labs, 0, 4)

groupPer<-stmetlsP%>%group_by(sm_lab, VT)%>%summarise(RA=mean(RA)*100, counts=n())

interest<-paste0("VTX", c("00222", "00115", "00132", "00063", "00092"))
interest
groupPer%>%subset(VT%in%interest)%>%subset(sm_lab!="Ca_C" & sm_lab!="Ca_O")%>%arrange(VT)

#meltmutate(diseaserav, VT, DiseaseRating)
#drmeltsg<-psmelts%>%arrange(desc(RA))


#meltmutate(mangmntrav, VT, managment)
#mgmeltsg<-psmelts%>%arrange(desc(RA))

head(stmeltsg)
head(stmetlsP)
topV<-c(
	as.character(unique(stmeltsg$VT)[1:20]))
#	as.character(unique(drmeltsg$VT)[1:15]),
#	as.character(unique(mgmeltsg$VT)[1:15]))
length(unique(topV))
topV


class(stmeltsg$VT)

#stmeltsg<-stmeltsg%>%arrange(labs)

#ggbarplot(stmeltsg, x="labs", y="RA", fill="VT")+themesizes



vtother<-function(dfs, group){
      subset_dfs<-dfs%>%subset(VT%in%unique(topV))
      other<-subset_dfs%>%group_by({{group}})%>%summarise(core=sum(RA), RA=1-core)
	print(other)
	other$VT<-"Other"
	vttible<<-full_join(subset_dfs, other)
}

topV
#Management
#vtother(mgmeltsg, managment)
#mg_vt<-vttible
#State
summary(sample_data(stmeltsg))
vtother(stmeltsg, sample)
st_vt<-vttible
vttible

#For Prevelance Graph
vtother(stmetlsP, labs)
st_P<-vttible
dim(st_P)
#DR
#vtother(drmeltsg, DiseaseRating)
#dr_vt<-vttible

length(unique(topV))
#head(dr_vt)





#VT Color Pallette
library('Polychrome')
set.seed(42)
topV<-c(topV, "Other")
vtpal<-createPalette(length(unique(topV)),"#05f2de", M=1500)
        show_col(vtpal)
topV<-data.frame(unique(topV))
colnames(topV)<-"VT"
dim(topV)


levels(codnames$CodeName)
topV<-inner_join(topV, codnames)
dim(topV)
colnames(topV)
length(unique(topV$VT))
length(vtpal)


sort(topV$CodeName)

names(vtpal)<-unique(topV$CodeName)
vtpal["Other"]<-"#D8D8D8"

vtpal		#Taxa Bar Plots Genus Level
other<-data.frame("Other", "Other")
names(other)<-c("VT", "CodeName")
rownames(other)<-"Other"
nrow(codnames)
codnames<-codnames[1:44,]


codnames<-rbind(codnames, other)

#mg_vt<-inner_join(mg_vt, codnames)
#sort(mg_vt$CodeName)

head(topV)
#head(mg_vt)
head(st_vt)
length(intersect(names(vtpal), codnames$CodeName))

#mg_vt%>%subset(CodeName=="Uk_130" )

show_col(vtpal)
#mavmeltbrplot<-ggbarplot(mg_vt, x="managment", y="RA", fill="CodeName")+
#		labs(x='', y="Relative Abundance", fill="Genus")+
#		themesizes+scale_y_continuous(labels=scales::percent)+
#		scale_fill_manual(values=vtpal)+theme(legend.position="none")
#mavmeltbrplot
st_vt<-inner_join(st_vt, codnames)
dim(st_P)
st_P<-inner_join(codnames,st_P)
colnames(st_P)

#Extract Genus
st_vt$Genus<-substr(st_vt$CodeName, 0,2)
st_P$Genus<-substr(st_P$CodeName, 0, 2)

gnpal<-createPalette(length(unique(st_vt$Genus)),"#FC4E07")
show_col(gnpal)
names(gnpal)<-unique(st_vt$Genus)

unique(st_vt$Genus)
unique(st_vt$CodeName)
#st_vt$group<-substr(st_vt$labs, 0, 4)
#table(st_vt$group)

colnames(st_vt)
stvmeltbrplot<-ggbarplot(st_vt,x='sample', y="RA", fill="CodeName")+ 
	labs(x='', y="Relative Abundance", fill="Virtual Taxa")+
	themesizes+scale_y_continuous(labels=scales::percent)+
                scale_fill_manual(values=vtpal)+theme(legend.position="bottom")+
		theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(),
			legend.box.margin=margin(6,6,6,6), legend.position='bottom')+
			guides(fill=guide_legend("Virtual \n Taxa", ncol=3))

stvmeltbrplot
vars(st_vt$Genus)

st_vt

at_Genus<-ggbarplot(st_vt,x='sample', y="RA", fill="Genus")+
	labs(x='', y="Relative Abundance", fill="Genus")+
		themesizes+scale_y_continuous(labels=scales::percent)+
                scale_fill_manual(values=gnpal)+
		theme(legend.position="bottom",legend.box="vertical", axis.text.x=element_blank(), 
		      axis.ticks.x=element_blank())
at_Genus
		gentab<-st_vt%>%group_by(Genus)%>%summarize(RA=sum(RA)*100)
gentab


st_vt%>%arrange(desc(RA))

fig4_ab<-ggarrange(at_Genus, stvmeltbrplot, labels=c("a", "b"), nrow=2, font.label=list(size=18))


fig4_ab


#CD are prevalence Histographs Stack Barplots
head(st_P)

st_P$group<-substr(st_P$labs, 0,4)
st_P$prev<-ifelse(st_P$RA>0, 1,0)

table(st_P$group)
prev_tab<-st_P%>%group_by(CodeName, Genus, group)%>%subset(Genus!="Ot")%>%
	summarize(RA=sum(RA), prev=sum(prev), counts=n())%>%subset(prev>0)%>%
	arrange(desc(prev))
prev_tab
prev_tabAll<-st_P%>%group_by(CodeName, Genus)%>%subset(Genus!="Ot")%>%summarise(RA=sum(RA), 
				prev=sum(prev), counts=n())%>%subset(prev>0)%>%arrange(desc(prev))

prev_tabAll$group<-"Total"
prev_tab<-full_join(prev_tab, prev_tabAll)

table(prev_tab$group)

namesG<-c("Ca_C", "Ca_O", "Fl_1", "Fl_2", "Fl_3", "Fl_4", "Fl_5", "Total")
colsG<-c(cols[c(2,5,17,15,13,11,9)], "black")
names(colsG)<-namesG
show_col(colsG)
vt_prev<-prev_tab%>%subset(group!="Total")%>%ggbarplot(x="CodeName", y="prev", fill="group", 
		   label=TRUE, lab.col="black", lab.vjust=1.25, lab.size=3)+
		scale_fill_manual(values=colsG)+themesizes

vt_prev

Gprev_tab<-st_P%>%group_by(group,labs, Genus)%>%summarize(RA=sum(RA),counts=n())
Gprev_tab$prev<-ifelse(Gprev_tab$RA>0,1,0)
Gprev_tab2<-Gprev_tab%>%group_by(group, Genus)%>%summarize(RA=sum(RA), prev=sum(prev), counts=n())%>%
		subset(Genus!="Ot")%>%subset(prev>0)%>%arrange(desc(prev))
which(Gprev_tab2$prev==0)
gen_prev<-
	ggbarplot(Gprev_tab2, x="Genus", y="prev", fill="group", 
		    label=TRUE, lab.col="black",  lab.vjust=1.25, lab.size=3.4)+
		scale_fill_manual(values=colsG)+themesizes

gen_prev_dot<-ggdotchart(Gprev_tab2, x="Genus", y="prev", color="group", dot.size=5,label=TRUE, rotate=TRUE, sorting="descending")+
	scale_fill_manual(values=colsG)+theme_cleveland()

vt_prev_dot<-prev_tab%>%subset(group=="Total")%>%ggdotchart(x="CodeName", y="prev", color="group", 
			dot.size="prev",label=TRUE, rotate=TRUE, sorting="descending")+
	scale_color_manual(values=colsG)+theme_cleveland()+
	facet_grid(rows=vars(Genus), scales='free')+
	guides(size=FALSE)+labs(y="Prevelance (Total n=69)")

vt_prev_dot
fig4_alt<-ggarrange(stvmeltbrplot+theme(legend.position="top")+guides(fill=guide_legend(nrow=2, byrow=TRUE)),
		    vt_prev_dot)

fig4_alt
fig4_cd<-ggarrange(gen_prev_dot,vt_prev_dot, labels=c("c", "d"), font.label=list(size=18), common.legend=TRUE, nrow=2)

fig4_ab<-ggarrange(at_Genus, stvmeltbrplot, labels=c("a", "b"), nrow=2, font.label=list(size=18))
fig4abcd<-ggarrange(ggarrange(at_Genus, stvmeltbrplot, labels=c('a', 'b'), nrow=2, font.label=list(size=18)),
			      vt_prev_dot+theme(legend.position='none'),ncol=2, labels=c("", "c"), 
			      widths=c(2,1 ),font.label=list(size=18), hjust=1.25)
fig4abcd
#	dr_vt<-inner_join(dr_vt, codnames)

#Taxa Bar Plot and Prev Diagram

ggsave(paste0(g, "Figure2_Taxa_hist.png"), device='png', width=80, height=100, units='mm', scale=2.5)
ggsave(paste0(g, "Figure2_Taxa_hist.tiff"), device='tiff', compression="lzw", dpi=600)



#drvmeltbrplot<-ggbarplot(dr_vt, x="DiseaseRating", y="RA", fill="CodeName")+
#		labs(x='', y="Relative Abundance", fill="Virtual Taxa")+
#		themesizes+scale_y_continuous(labels=scales::percent)+
 #               scale_fill_manual(values=vtpal)+theme(legend.position="right")
sort(dr_vt$CodeName)
	#names(vtpal)[13]<-"Dk_301"
names(vtpal)
ggarrange(ggarrange(stvmeltbrplot, mavmeltbrplot,nrow=2, labels=c("a", "b"), font.label=list(size=18)),
	  drvmeltbrplot, labels=c("", "c"), font.label=list(size=18), common.legend=TRUE, ncol=2)

#ggsave(paste0(g, 'Fig4_Taxa_Bar.png'))

#ggsave(paste0(g, "Figure4_Taxa_Bar.tiff"), device='tiff', compression="lzw", dpi=600)


#Stats Kruskall Wallis with Dunn or Tukey Test 
library('broom')


mg_vt%>%group_by(VT)%>%do(tidy(kruskal.test(x=.$RA, g=.$managment)))%>%arrange(p.value)
#Nothing Significant in management 
dr_vt%>%group_by(VT)%>%mutate(RA=RA*100)%>%do(tidy(kruskal.test(x=.$RA, g=.$DiseaseRating)))%>%arrange(p.value)
#Nothing Significant In Disease Rating
st_vt%>%group_by(VT)%>%do(tidy(kruskal.test(x=.$RA, g=.$State)))%>%arrange(p.value)
#Nothing Significant at State Levelt

#By Genus 

#Management 
mg_vt$Genus<-sapply(strsplit(as.character(mg_vt$CodeName), "_"), "[", 1)
mg_g<-mg_vt%>%mutate(Genus=replace(Genus, Genus=='DK', 'Dk'))%>%
		group_by(Genus, managment)%>%summarize(RA=sum(RA)*100)
mg_g%>%do(tidy(kruskal.test(x=.$RA, g=.$managment)))%>%arrange(p.value)


#States 
st_vt$Genus<-sapply(strsplit(as.character(st_vt$CodeName), "_"), "[", 1)
st_vt
st_g<-st_vt%>%mutate(Genus=replace(Genus, Genus=='DK', 'Dk'))%>%
		group_by(Genus, State)%>%summarize(RA=sum(RA)*100)
st_g%>%do(tidy(kruskal.test(x=.$RA, g=.$State)))%>%arrange(p.value)

#Disease Rating

dr_vt$Genus<-sapply(strsplit(as.character(dr_vt$CodeName), "_"), "[", 1)
head(dr_vt)
dr_g<-dr_vt%>%mutate(Genus=replace(Genus, Genus=='DK', 'Dk'))%>%
		group_by(Genus, DiseaseRating)%>%summarize(RA=sum(RA)*100)
dr_g%>%do(tidy(kruskal.test(x=.$RA, g=.$DiseaseRating)))%>%arrange(p.value)


```
#Tax Table
```{r}
#taxdf<-as.data.frame(tax_table(amf))
#taxdf<-taxdf[c("Row.names", "Order", "VT")]
#dim(taxdf)
#colnames(taxdf)<-c("ASV", "Genus", "VT")
#write.csv(taxdf, "TaxaDF_304.11.09.2020")
#colnames(taxdf)
#write.csv(taxdf%>%dplyr::count(VT, Genus), "11.09.2020_VT_GENUS_Counts.csv")

```
#Figure 3: Beta Diversity 
```{r}

#Bray Curtis NMDS
DR
states
mang
DRbray<-ordinate(DR, "NMDS", "bray")
statesbray<-ordinate(states, "NMDS", "bray")
mangbray<-ordinate(mang, "NMDS", "bray")


#DiseaseRating Beta Diversity 
DRbeta<-plot_ordination(DR, DRbray,color="DiseaseRating")+
	geom_point(size=5)+
	theme_classic()+
	geom_vline(xintercept=0, colour='grey70')+
	geom_hline(yintercept=0, color='grey70')+
	scale_color_manual(values=rev(cols[seq(9,18,2)]))+
	scale_size_manual(values=c(3,6))+
	scale_shape_manual(values=c(1,16),name="Year", labels=c("Year1", "Year2"))+	
	themesizes+
	annotate(geom="text", x=-1.0, y=1,
		 label=as.character(expression("R^{2} == 0.1949")),parse=T)+
		annotate(geom="text", x=-1.0, y=.9, label="p=0.018")
DRbeta

#State Beta Diverstiy
StateBeta<-
	plot_ordination(states, statesbray,color="State")+
	geom_point(size=5)+
	theme_classic()+
	geom_vline(xintercept=0, colour='grey70')+
	geom_hline(yintercept=0, color='grey70')+
	scale_color_manual(values=cols[c(6,8)])+
	annotate(geom="text", x=-2.8, y=.65,
		 label=as.character(expression("R^{2} == 0.0698")),parse=T)+
		annotate(geom="text", x=-2.8, y=.6, label="p=0.001")+
	themesizes

#Management Beta
mangeBeta<- 
	plot_ordination(mang, mangbray,color="managment")+   
	geom_point(size=5)+
	theme_classic()+
	geom_vline(xintercept=0, colour='grey70')+
	geom_hline(yintercept=0, color='grey70')+
	scale_color_manual(name="Management",values=cols[c(2,4,5)])+
	themesizes+
	annotate(geom="text", x=-3.0, y=1.1,
		 label=as.character(expression("R^{2} == 0.0807")),parse=T)+
		annotate(geom="text", x=-3.0, y=1, label="p=0.001")

ggarrange(ggarrange(StateBeta, mangeBeta, labels=c("a", "b"),font.label=list(size=18), ncol=2),
	 DRbeta, labels=c("","c"),nrow=2, font.label=list(size=18))

ggsave(paste0(g, "Betadiv.png"), device='png', width=80, units='mm', height = 100, scale=3)
ggsave(paste0(g, "Figure3_Beta.tiff"), device='tiff', compression="lzw", dpi=600)
```

#Hypothesis testing: differences in Centroids 
```{r}
set.seed(1)

#Make Maps
	
dmap<-data.frame(sample_data(DR)) 
smap<-data.frame(sample_data(states))
mmap<-data.frame(sample_data(mang))

#Bray Distance
dbray<-phyloseq::distance(DR, method="bray")
sbray<-phyloseq::distance(states, method="bray")
mbray<-phyloseq::distance(mang, method="bray")

#Adonis
set.seed(28)
adonis(dbray~DiseaseRating, data=dmap, method='bray')
adonis(sbray~State, data=smap, method='bray')
adonis(mbray~managment, data=mmap, method='bray')

set.seed(52)

#Pairwise Perm Manova

library("RVAideMemoire")
pairwise.perm.manova(dbray, dmap$DiseaseRating)
pairwise.perm.manova(sbray, smap$State)
pairwise.perm.manova(mbray, mmap$managment)

betadr<-betadisper(dbray, dmap$DiseaseRating)
betast<-betadisper(sbray, smap$State)
betama<-betadisper(mbray, mmap$managment)

anova(betadr)
anova(betast)
anova(betama)
betama
```

#Figure 4: Tissue Venn Diagram 

```{r}
myplot<-function(mydata,x,y){
ggdotchart(sta3, x="VT", y="prev")
t_prev<-ggbarplot(mydata,x="VT", y='prev', color='color',group='colpat',

	add.params= list(color='color',size=2))+
	theme(strip.background=element_blank(), 
	strip.text.x=element_blank())+
        rotate_x_text(45, vjust=1)+
	labs(x=' ', y='Relative Abundance (%)')+
	scale_y_continuous(labels=function(x) x*100)+
	scale_color_identity()+
	theme(axis.text.x=element_text(angle=90, hjust=1,vjust=.45, size=12),
        axis.text.y=element_text(size=8),axis.title.y=element_text(size=10),
        legend.position='none', plot.margin=margin(0,2,2,0,"mm"))



high
mydata3<-mydata%>%subset(fac!="High")
	low<-
ggdotchart(mydata3,x="VT", y=y, color='color',group='colpat',
	dot.size=7,add="segments", 
	label=round((mydata3[,y]*100),0),
	font.label=list(color="white", size=7, vjust=0.5),
	add.params= list(color='color',size=2))+
	theme(strip.background=element_blank(), 
	strip.text.x=element_blank())+
        rotate_x_text(45, vjust=1)+
	labs(x=' ', y='')+
	scale_y_continuous(labels=function(x) x*100)+
	scale_color_identity()+
 	theme(axis.text.x=element_text(angle=90, hjust=1,vjust=.45, size=12), 
	axis.text.y=element_text(size=8),axis.title.y=element_text(size=10),
	legend.position='none', plot.margin=margin(0,2,2,0,"mm"))



plot<-grid.arrange(arrangeGrob(high, low, widths=c(widths1,widths2)))
}

```

#For Venn Diagram Agglomerate to VT level then transform Rel abund 
```{r}

vtglom<-function(ps2change){
	td<-data.frame(tax_table(ps2change))
	td<-td[,c("Kingdom","CodeName")]
	changedps<-ps2change
tax_table(changedps)<-as.matrix(td)
	print(ps2change)
	print(changedps)
	changedps<-tax_glom(changedps, "CodeName")
	print(changedps)
	changedps<-transform_sample_counts(changedps, function(x) x/sum(x))
	print(sample_sums(changedps))
	pswant<<-changedps
}
vtglom(states)	
statesrav<-pswant

vtglom(DR)	
diseaserav<-pswant

vtglom(mang)	
mangrav<-pswant







```


#Classic Venn Diagrams for Pairs (States and Management) At Taxa level
```{r}

ca<-subset_samples(statesrav, State=="California")
fl<-subset_samples(statesrav, State!="California")
cn<-subset_samples(mangrav, managment=="Conventional")
or<-subset_samples(mangrav, managment=="Organic")

subset_taxa(ca, taxa_sums(ca)>0)
subset_taxa(fl, taxa_sums(fl)>0)
subset_taxa(cn, taxa_sums(cn)>0)
subset_taxa(or, taxa_sums(or)>0)

caTax<-transform_sample_counts(merge_samples(fl, "State"), function(x) x/sum(x)*100)
caTax<-psmelt(caTax)
colnames(caTax)
caTax<-caTax[,c(3,13,25, 20)]
sum(caTax$Abundance)
interest<-c("Dk_222", "Rz_113", "Rz_115", "Uk_132", "Sg_063")
caTax%>%subset(CodeName%in%interest)
head(caTax%>%arrange(desc(Abundance)), n=10)


#Remove taxa that aren't prevalence
#prevalent threshold

thres=.1
ntax=.01

library('microbiome')
ca<-core(ca, detection=ntax,prevalence=thres)
fl<-core(fl, ntax, thres)
cn<-core(cn, ntax, thres)
or<-core(or, ntax, thres)
ca
fl
cn
or

#Dataframes for Easier manipulations at Taxa Level Will Also Do VT level
caT<-as.character(data.frame(tax_table(ca))$Row.names)
caV<-as.character(data.frame(tax_table(ca))$CodeName)

flT<-as.character(data.frame(tax_table(fl))$Row.names)
flV<-as.character(data.frame(tax_table(fl))$CodeName)

cnT<-as.character(data.frame(tax_table(cn))$Row.names)
cnV<-as.character(data.frame(tax_table(cn))$CodeName)


orT<-as.character(data.frame(tax_table(or))$Row.names)
orV<-as.character(data.frame(tax_table(or))$CodeName)


#Virtual Taxa
#Find Inersections
caVa<-length(unique(caV))
flVa<-length(unique(flV))
cnVa<-length(unique(cnV))
orVa<-length(unique(orV))

#Remove NAs
#aV<-caV[!is.na(caV)]
#lV<-caV[!is.na(flV)]
#nV<-caV[!is.na(cnV)]
#rV<-caV[!is.na(orV)]

caflV<-intersect(caV, flV)

#setdiff(orV, cnorV)
cnorV<-intersect(cnV, orV)
caV
#Pairwise Venn for States
svV<-venn.diagram(list(California=caV, Florida=flV),
        lty="blank",
         fill=cols[c(8,6)], alpha=.75, 
        cex=1.4, cat.fontface=2, 
        cat.cex=0, 
      cat.just=list(c(4, .5), c(-4, .5)), 
	filename=NULL)
svV[[5]]$label<-paste(setdiff(caV,flV), collapse="\n")
svV[[6]]$label<-paste(setdiff(flV, caV), collapse="\n")
svV[[7]]$label<-paste(intersect(caV, flV), collapse="\n")

grid.newpage()
png(paste0(g, "VennDiaStateVT.png"))
grid.draw(svV)
dev.off()



mvV<-venn.diagram(list(Conventional=cnV, Organic=orV),
        lty="blank",
         fill=cols[c(2,5)], alpha=.75, 
        cex=1.4, cat.fontface=2, 
        cat.cex=0, 
       cat.just=list(c(2, .5), c(-4, .5)), 
	filename=NULL)
mvV[[5]]$label<-paste(setdiff(orV,cnV), collapse="\n")
mvV[[6]]$label<-paste(setdiff(cnV, orV), collapse="\n")
mvV[[7]]$label<-paste(intersect(cnV, orV), collapse="\n")

grid.newpage()
#png(paste0(g, "VennDiamangVT.png"))
grid.draw(mvV)
#dev.off()


```
#Disease Rating UpsetR Graph
```{r}
sample_sums(diseaserav)
subset_samples(diseaserav, taxa_sums(diseaserav)>0)
DR_VD<-psmelt(diseaserav)
unique(DR_VD$CodeName)


#Add Presence And Abscence
	combo<-DR_VD


	combo$pres<-ifelse(combo$Abundance>0,1,0)
	print(sum(combo$pres))
        print(sum(combo$Abundance))
	diseaserav
sta<-combo%>%group_by(CodeName, DiseaseRating)%>%
	summarize(counts=n(), prev=sum(pres)/n()*100, bin=ifelse(prev>10,1,0),
	incidence=sum(pres), abund=mean(Abundance))


sta2<-sta%>%spread(DiseaseRating, bin)
vts<-sta2$CodeName

	
sta2<-sta2%>%mutate(`1`=replace_na(`1`, 0), `2`=replace_na(`2`, 0),
        `3`=replace_na(`3`, 0), `4`=replace_na(`4`,0),`5`=replace_na(`5`, 0))
head(sta2)
range(sta2$prev)
diseaserav
sta2<-sta2%>%dplyr::group_by(CodeName)%>%dplyr::summarize(counts=sum(counts),
        DR1=sum(`1`),DR2=sum(`2`), DR3=sum(`3`),DR4=sum(`4`),DR5=sum(`5`),
		 RA=mean(abund),
                prev=sum(incidence), 
		colpat=sum(ifelse(DR1>0,1,0),ifelse(DR2>0,2,0),
                 ifelse(DR3>0,5,0), ifelse(DR4>0,10,0),ifelse(DR5>0,20,0)))


sta2$inc<-(sta2$prev/sta2$counts)

sta3<-sta2%>%subset(colpat>0)

sta3<-sta3%>%subset(RA>=.01)
dim(sta3)
dim(sta2)
sta3$fac<-ifelse(sta3$RA>.08, "High", "Low")
table(sta3$fac)
table(sta3$colpat)
unique(sta3$CodeName)
sta3<-data.frame(sta3)

library(UpSetR)
library(stringr)

ppi<-600
library("ggsci")

upsetpal<-c(pal_aaas('default', alpha=.8)(10), pal_rickandmorty('schwifty')(10))
show_col(upsetpal)
head(sta2)


#tiff(paste0(g, "venn_EndoTis.tiff"),width=6.6*ppi, height=6.6*ppi, res=ppi)


colnames(sta3)[8]<-"RelativeAbundance"

#At ASV Level
widths1=1.5
widths2=2.5
colnames(sta3)
head(fromList(sta3))
#png(paste0(g, "DiseaseRatingVennDiagram.png"))




sets<-names(sta3[3:7])
metadata<-data.frame(cbind(sets, as.numeric(as.character(table(sample_data(DR)[,"DiseaseRating"])))))
names(metadata)<-c("Sets", "NGroup") 
metadata$NGroup<-as.numeric(as.character(metadata$NGroup))



## Now make the plot
mydata=sta3
x="VT"
y="prev"
color="black"
plot1<-function(mydata,x, y){
	ggdotchart(mydata,x="VT", y=y, color='color',group='colpat',
	dot.size=4,add="segments", 


	add.params= list(color='color',size=2))+
	theme(strip.background=element_blank(), 
	strip.text.x=element_blank())+
	labs(x=' ', y='Prevalence')+
	scale_color_identity()+
 	theme(axis.text.x=element_text(angle=90, hjust=1,vjust=.45, size=10), 
	axis.text.y=element_text(size=8),axis.title.y=element_text(size=10),
	legend.position='none', plot.margin=margin(0,2,2,0,"mm"))

}


#	VT_prv<-
#		ggplot(data=mydata,aes_string(x=x, y=y, colour='color'))+geom_bar(stat='identity')+
#
#		labs(x="", y="Prevalence")+themesizes+scale_fill_identity()+
#		theme(axis.text.x=element_text(size=10, angle=45))
#
#		}


require('ggrepel')
library(ggforce)
colnames(sta3)
mydata=sta3

x='prev'
y='RelativeAbundance'
head(mydata)
table(mydata$colpat)
class(mydata$colpat)
mydata%>%subset(colpat=='20')

hardcols<-1:length(mydata$colpat)
for (i in 1:length(mydata$colpat)){
	print(mydata$colpat[i])
	if (mydata$colpat[i]==1 ){
		hardcols[i]=cols[17]
	
	}else if (mydata$colpat[i]==2 ){
		hardcols[i]=cols[15]

	}else if (mydata$colpat[i]==8 ){
		hardcols[i]=cols[18]
	
	}else if (mydata$colpat[i]==10 ){
		hardcols[i]=cols[11]
	
	} else if (mydata$colpat[i]==11 ){
		hardcols[i]=upsetpal[9]
	
	}else if (mydata$colpat[i]==16){
		hardcols[i]=upsetpal[10]

	} else if (mydata$colpat[i]==20| mydata$colpat[i]==30 ){
		hardcols[i]=cols[9]

	}else if (mydata$colpat[i]==28 | mydata$colpat[i]==38 ){
		hardcols[i]=upsetpal[4]
		
	} else {
		hardcols[i]=cols[10]}
	print(hardcols[i])
}

show_col(upsetpal)
show_col(cols)
table(mydata$hardcols, mydata$colpat)
class(hardcols)
mydata$hardcols<-hardcols
sta3$hardcols<-hardcols
#plot2<-function(mydata,x, y){
 	scatters<-
		ggplot(data=mydata,aes_string(x=x, y=y))+geom_point(size=3, aes(colour=hardcols))+themesizes+theme_classic()+
			labs(x="Prevalence", y="Relative Abundance")+scale_y_continuous(labels=scales::percent_format(accuracy=1))+
			scale_color_identity()+themesizes+
		theme(axis.text.x=element_text(angle=0), legend.position='bottom')+
		geom_label_repel(aes(label=CodeName, colour=hardcols))+ylim(0,
#		facet_zoom(ylim=c(0,.05),zoom.data=ifelse(RelativeAbundance<=.05, NA, FALSE))
#}
themesizes

ggsave(paste0(g, "UpsetGraphB.png"),units='mm', width=100,height=100, dpi=600, scale=1)
scatters

#attributeplots <- list(gridrows = 100,
 #                 plots = list(list(plot = plot2, x = "prev", y = "RelativeAbundance", queries = TRUE)))
                 

show_col(upsetpal) 
#png('drupsetr_11.11.2020.png')


class(metadata$NGroup)
table(sta3$hardcols, sta3$colpat)
sta3

upset(sta3,sets=c("DR5", "DR4", "DR3", "DR2", "DR1"),keep.order=TRUE, 
      order.by='freq',point.size=4,text.scale=(c(2.0,2.0,1.5,1.5,2,2.0)),
      set.metadata = list(data = metadata, plots = list(list(type = "hist", 
    column = "NGroup", colors=cols[seq(9,18,2)],assign = 20))),






#      attribute.plots=attributeplots,
        queries = list(
		list(query = intersects, params = list("DR1"), color = cols[17], active = T),
#        	list(query = intersects, params = list("DR2"), color = cols[15], active = T),
#        	list(query = intersects, params = list("DR4"), color = cols[11], active = T),
        	list(query = intersects, params = list("DR5"), color = cols[9], active = T),

#		list(query = intersects, params = list("DR1",  "DR4"),color = upsetpal[9], active = T),
#		list(query = intersects, params = list("DR5",  "DR4"),color = cols[10], active = T),

	        list(query = intersects, params = list("DR1", "DR2", "DR3"), color = cols[18], active = T),
	        list(query = intersects, params = list("DR1", "DR3", "DR4"), color = upsetpal[10], active = T),
	        list(query = intersects, params = list("DR1", "DR2", "DR3", "DR5"), color = upsetpal[4], active = T),
	        list(query = intersects, params = list("DR1", "DR2", "DR3","DR4","DR5"), color = upsetpal[4], active = T)))

#dev.off()

head(sta3)
```

#Differential Abundance (Deseq2)



```{r}

vtpal2<-c(vtpal,vtpal,"#000000")
length(vtpal2)
plot_bar(amf3, fill="VT" )+scale_fill_manual(values=vtpal2)+theme_classic()+
		facet_grid(managment+DiseaseRating~State, scale="free_x", drop=TRUE)

amf3
#Define Groups
#Run Deseq
	
 	gm_mean = function(x, na.rm=TRUE){
		  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))}


deseq2fun<-function(diagdds,group,thresh, ps){

		geoMeans = apply(counts(diagdds), 1, gm_mean)
		diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
		diagdds = DESeq(diagdds, fitType="local")
		keeptit<<-diagdds

}

```


```{r}
#States
	#statesG<-tax_glom(states, "Order")

	#ab_deseq=phyloseq_to_deseq2(statesG, ~State)
	#deseq2fun(ab_deseq, "states", 0.01, statesG)

	#State_resG<-keeptit
 	statesV<-states
	mangV<-mang
mangV
	tax_table(statesV)<-tax_table(statesV)[,c("Superkingdom","CodeName")]
   tax_table(mangV)<-tax_table(mangV)[,c("Superkingdom","CodeName")]

	
#	statesV<-tax_glom(statesV, "CodeName")

	ab_deseq=phyloseq_to_deseq2(statesV, ~State)

   deseq2fun(ab_deseq, "states", 0.01, statesV)

	State_resV<-keeptit

   #Management
	#mangG<-tax_glom(mang, "Order")

	#ab_deseq=phyloseq_to_deseq2(mangG, ~managment)
	#deseq2fun(ab_deseq, "managment", 0.01, mangG)

	#mange_resG<-keeptit
	#mangV<-tax_glom(mangV, "CodeName")
	mangV
	ab_deseq=phyloseq_to_deseq2(mangV, ~managment)
	deseq2fun(ab_deseq, "managment", 0.01, mang)

	mange_resV<-keeptit


#Deseq2 Virtual Taxa Management Level

ds_results<- mange_resV
ps_obj<- mangV
groupQ<-"managment"

       st_DS2<-data.frame(results(ds_results,
                name=resultsNames(ds_results)[2]))

        st_DS3<-cbind(as(st_DS2, "data.frame"),
                as(tax_table(ps_obj)[rownames(st_DS2), ], "matrix"))

        st_DS3<-st_DS3%>%subset(padj<0.01)

        AB_deseq3<-st_DS3%>%subset(log2FoldChange >1 | log2FoldChange< -1)
        print(dim(st_DS3))
        print(dim(AB_deseq3))

colnames(AB_deseq3)
hist(AB_deseq3$log2FoldChange)
sort(unique(as.character(AB_deseq3$CodeName)))
AB_deseq3$seqname<-make.unique(as.character(AB_deseq3$CodeName))
print(AB_deseq3$seqname)
#write.csv(st_DS3, paste0(g, "mangementVT_deseq2.csv"))
                        #Merge By Group
         ab<-merge_samples(ps_obj, groupQ)
        ab<-transform_sample_counts(ab, function(x) (x/sum(x)*100))
        print(sample_sums(ab))
                #Remove Low RA 
       ab<-subset_taxa(ab, taxa_sums(ab)>1)
       ab

 #Melt Phyloseq
        abb<-psmelt(ab)

        colnames(abb)[1]<-"ASVs"
        abb<-abb%>%
                group_by(ASVs, CodeName, Sample)%>%
        summarize(Abundance=mean(Abundance))%>%spread("Sample", "Abundance")
        print(colnames(abb))

                #Subset Based on Genera from Deseq2
        AB_deseq3$ASVs<-rownames(AB_deseq3)
        abhm<-abb%>%subset(ASVs%in%AB_deseq3$ASVs)

        print(dim(abhm))
	abhm
	print(dim(AB_deseq3))


        abra_l2f<-inner_join(abhm, AB_deseq3)

        print(dim(abra_l2f))
        abra_l2f<<-abra_l2f%>%arrange(log2FoldChange)

abra_l2f$seqname<-make.unique(as.character(abra_l2f$CodeName))
abra_l2f$label<-ifelse(abra_l2f$log2FoldChange>0, "Organic", "Conventional")
abra_l2f$RA<-round(pmax(abra_l2f$Organic, abra_l2f$Conventional),0)
abra_l2f$RA


print(abra_l2f$seqname)
print(abra_l2f, n=21)

sigtab<-abra_l2f
sigtab[,c(2:12)]
opA<-ggplot(sigtab, aes(x=CodeName, y=log2FoldChange, color=label)) + geom_point(aes(size=RA)) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+theme_classic()+
  scale_color_manual(values=cols[c(2,5)])+labs(x='')+geom_hline(yintercept=0, size=1)

opB<-ggbarplot(sigtab, x="seqname", y="log2FoldChange", fill="label", position=position_dodge(0.9)) + geom_point(aes(size=RA), shape=2) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+theme_classic()+
  scale_fill_manual(values=cols[c(2,5)])+labs(x='')+geom_hline(yintercept=0, size=1)+coord_flip()

ggarrange(opA, opB, nrow=2)

#hmMV<-data.frame(abra_l2f)
#rownames(hmMV)<-hmMV$seqname
 #       colnames(hmMV)
#	logabund<-hmMV[,c("Conventional","Organic","log2FoldChange")]
   #    logabund<-as.matrix(logabund)
  #       head(logabund)
 #       summary(logabund)

#colnames(hmMV)


#lwfc_hm<-Heatmap(logabund[,3], col=colorRamp2(c(-3, 0, 30), c(cols[20],'grey90',cols[8])),
#		 name="L2FC", show_column_dend=FALSE, show_row_dend=FALSE,
#		 width=unit(.25, 'cm'), height=unit(10, 'cm'),cluster_rows=FALSE,
#		 cluster_columns=FALSE, row_names_side="left",
 #               row_names_gp=gpar(fontsize=12), column_names_gp=gpar(fontsize=12))
#show_col(cols)
#draw(lwfc_hm)
#heatmapcols<-viridis_pal()(6)
#show_col(heatmapcols)
#lhm<-Heatmap(logabund[,1:2], col=colorRamp2(c(0,1,2.5,5,10,20, 30),
#	c("black",heatmapcols[1:6])),
#	 name="Relative\nAbundance\n",
 #       show_column_dend=FALSE, show_row_dend=FALSE,
#	 width=unit(1, 'cm'),height=unit(10,'cm'),cluster_rows=FALSE,
 #               column_order=c("Conventional", "Organic"),
#	 cluster_columns=FALSE, row_names_side="left",
#                row_names_gp=gpar(fontsize=12), column_names_gp=gpar(fontsize=12))
#AB_HM<-lhm+lwfc_hm
#draw(AB_HM, gap=unit(0, 'mm'))


#png(paste0(g, "Managment_VT.png"))
#draw(AB_HM, gap=unit(0, 'mm'))
#dev.off()

#States Virtual Taxa Level Heatmap
library('viridis')
ds_results<- State_resV
ps_obj<- statesV
groupQ<-"State"
ps_obj
       st_DS2<-data.frame(results(ds_results,
                name=resultsNames(ds_results)[2]))

        st_DS3<-cbind(as(st_DS2, "data.frame"),
                as(tax_table(ps_obj)[rownames(st_DS2), ], "matrix"))

        st_DS3<-st_DS3%>%subset(padj<0.01)

        AB_deseq3<-st_DS3%>%subset(log2FoldChange >1 | log2FoldChange< -1)
        print(dim(st_DS3))
        print(dim(AB_deseq3))


                        #Merge By Group
        ab<-merge_samples(ps_obj, groupQ)
        ab<-transform_sample_counts(ab, function(x) (x/sum(x)*100))
        print(sample_sums(ab))
                #Remove Low RA 
        ab<-subset_taxa(ab, taxa_sums(ab)>1)
                #Melt Phyloseq
        abb<-psmelt(ab)

        colnames(abb)[1]<-"ASVs"
        abb<-abb%>%
                group_by(ASVs, CodeName, Sample)%>%
        summarize(Abundance=mean(Abundance))%>%spread("Sample", "Abundance")
        print(colnames(abb))

                #Subset Based on Genera from Deseq2
        AB_deseq3$ASVs<-rownames(AB_deseq3)
        abhm<-abb%>%subset(ASVs%in%AB_deseq3$ASVs)

        print(dim(abhm))
        print(dim(AB_deseq3))


        abra_l2f<-inner_join(abhm, AB_deseq3)

        print(dim(abra_l2f))
        abra_l2f<<-abra_l2f%>%arrange(log2FoldChange)
abra_l2f$seqname<-make.unique(as.character(abra_l2f$CodeName))
print(abra_l2f$seqname)

abra_l2f$seqname<-make.unique(as.character(abra_l2f$CodeName))
abra_l2f
abra_l2f$label<-ifelse(abra_l2f$log2FoldChange>0, "Florida", "California")
abra_l2f$RA<-round(pmax(abra_l2f$Florida, abra_l2f$California),0)
abra_l2f$RA


print(abra_l2f$seqname)
print(abra_l2f, n=21)

sigtab<-abra_l2f
sigtab[,c(2:12)]
opA<-ggplot(sigtab, aes(x=CodeName, y=log2FoldChange, color=label)) + geom_point(aes(size=RA)) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+theme_classic()+
  scale_color_manual(values=cols[c(6,8)])+labs(x='')+geom_hline(yintercept=0, size=1)

opB<-ggbarplot(sigtab, x="seqname", y="log2FoldChange", fill="label", position=position_dodge(0.9)) + geom_point(aes(size=RA), shape=2) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+theme_classic()+
  scale_fill_manual(values=cols[c(6,8)])+labs(x='')+geom_hline(yintercept=0, size=1)+coord_flip()

ggarrange(opA, opB, nrow=2)




hmSV<-data.frame(abra_l2f)
rownames(hmSV)<-hmSV$seqname
       colnames(hmSV)
	logabund<-hmSV[,c("California","Florida","log2FoldChange")]
      logabund<-as.matrix(logabund)
	dr3$lab<-paste0("DR3")
	dr4$lab<-paste0("DR4")
dr5$lab<-paste0("DR5")


	dim(dr1)
	dim(dr2)
	dim(dr3)
	dim(dr4)
	dim(dr5)
	avjoin<-rbind(dr1, dr2, dr3, dr4, dr5)
colnames(avjoin)
#
	avjoin_sig<-subset(avjoin, padj<0.01)
#
	print(dim(avjoin_sig))
	
	avjoin_sig3<-subset(avjoin_sig, log2FoldChange>1 | log2FoldChange< (-1))


	dim(avjoin_sig3)
	head(avjoin_sig3)
ggplot(avjoin_sig3, aes(x=log2FoldChange, y=CodeName, color=lab))+
		geom_vline(xintercept=0.0, color="#bababa", size=0.5)+
		geom_point(size=6)+ggtitle("DR")+
			theme(plot.title=element_text(hjust=0.5),axis.text.y=element_text(size=18))
	ggsave(paste0(g, "DR_deseq_Vplot", ".png"))

#	write.csv(avjoin_sig, paste0(g,"Bac_gen_glom_deseq_tissues", ".csv"))
	#avjoin_sig3<-read.csv("graphs/Bac_gen_glom_deseq_tissues.csv")



```
#Deseq2 Heat Map 
```{r}
dim(avjoin_sig3)

#Combine Bac and Fun

head(abfb)
colnames(combo)[1]<-"ASVs"

abfb<-combo%>%group_by(CodeName,DiseaseRating)%>%
	summarize(Abundance=mean(Abundance)*100)%>%spread("DiseaseRating", "Abundance")


abhm<-abfb%>%subset(CodeName%in%avjoin_sig3$CodeName)
dim(abhm)
dim(abfb)


colnames(abhm)
abhm$sums<-rowSums(abhm[,c(2:6)])
abra_l2f<-inner_join(abhm, avjoin_sig3)

hm4<-data.frame(abhm)
	
        rownames(hm4)<-hm4$CodeName
        colnames(hm4)
        logabund<-hm4[,c(2:6)]
 logabund<-as.matrix(logabund)
         head(logabund)
        summary(logabund)

colnames(hm4)




colnames(logabund)<-c("DR1", "DR2", "DR3", "DR4", "DR5")

lhm<-Heatmap(logabund[,1:5], col=colorRamp2(c(0,1,5,10,20,30,40),
		c("black",heatmapcols[1:6])), name="Rel.\nAbundance",
        show_column_dend=FALSE, show_row_dend=FALSE, width=unit(3, 'cm'),height=unit(10,'cm'),
			cluster_rows=FALSE,
                column_order=c("DR1", "DR2", "DR3", "DR4", "DR5"), 
		cluster_columns=FALSE, row_names_side="left",
                row_names_gp=gpar(fontsize=12), column_names_gp=gpar(fontsize=12))

draw(lhm)
tiff(paste0(g, "DR_DeseqHeatMap.tiff"),width=6.6*ppi, height=6.6*ppi, res=ppi)
draw(lhm)
dev.off()


```

```{r}
sessionInfo()

```
